### PR TITLE
[Merged by Bors] - feat: port Data.Nat.WithBot

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

**Comment by mo271** [comment]
no it doesn't depend on tauto anymore...

File: Mathlib/Data/Nat/WithBot.lean [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+! if you have ported upstream changes.
+-/
+
+import Mathlib.Data.Nat.Order.Basic
+import Mathlib.Algebra.Order.Monoid.WithTop
+import Aesop
+
+/-!
+# `with_bot ℕ`
+
```
Comment (dupuisf) [inline Mathlib/Data/Nat/WithBot.lean:18]:
```suggestion
# `WithBot ℕ`

```

File: Mathlib/Data/Nat/WithBot.lean [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Lemmas about the type of natural numbers with a bottom element adjoined.
+-/
+
+
+namespace Nat
+
+namespace WithBot
+
+theorem add_eq_zero_iff {n m : WithBot ℕ} : n + m = 0 ↔ n = 0 ∧ m = 0 :=
+  by
```
Comment (dupuisf) [inline Mathlib/Data/Nat/WithBot.lean:28]:
```suggestion
theorem add_eq_zero_iff {n m : WithBot ℕ} : n + m = 0 ↔ n = 0 ∧ m = 0 := by
```
This is the style we're going for. (And the same for the rest of the file)

Comment (mo271) [reply Mathlib/Data/Nat/WithBot.lean]: done

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**